src/libexpr/eval.cc

```diff
@@ -364,6 +364,36 @@ EvalState::EvalState(
 
 EvalState::~EvalState()
```

Comment:
Not super happy about doing this in a destructor, though it probably works fine.

Please have a look at `NIX_SHOW_STATS`. You could move this logic out of the destructor and into a method, and then call it in the same way.

Reply:
Currently it needs access to eval state to symbolize the stack, NIX_SHOW_STATS probably can access other global state.

Reply:
`NIX_SHOW_STATS` counters are also in `EvalState`.